src/rsz/src/Rebuffer.hh

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022-2025, The OpenROAD Authors
+
+#pragma once
+
+#include <vector>
+
+#include "db_sta/dbNetwork.hh"
+#include "db_sta/dbSta.hh"
+
+namespace rsz {
+
+using utl::Logger;
+
+using odb::Point;
+using sta::Corner;
+using sta::dbNetwork;
+using sta::Delay;
+using sta::Instance;
+using sta::LibertyCell;
+using sta::LibertyPort;
+using sta::MinMax;
+using sta::Net;
+using sta::Path;
+using sta::Pin;
+using sta::RiseFall;
+using sta::RiseFallBoth;
+using sta::Slack;
+using sta::Slew;
+using sta::Vertex;
+
+class Resizer;
+class BufferedNet;
+enum class BufferedNetType;
+using BufferedNetPtr = std::shared_ptr<BufferedNet>;
+using BufferedNetSeq = std::vector<BufferedNetPtr>;
+
+class Rebuffer : public sta::dbStaState
+{
+ public:
```

Comment:
Does this new Rebuffer class fit into @mguthaus's refactoring of rsz? Should rebuffer be a move or use Moves?

Reply:
> Does this new Rebuffer class fit into @mguthaus's refactoring of rsz?

Up to this point it's orthogonal, we coordinate

Reply:
> Should rebuffer be a move or use Moves?

It can be encapsulated as a move for repair_timing later on so that repair_timing can also benefit from this algorithm